elements/pl-parsons-problem/pl-parsons-problem.py

```diff
@@ -0,0 +1,328 @@
+import prairielearn as pl
+import lxml.html
+import random
+import base64
+import os
+DEFAULT_CHECK_INDENTATION = False
+
+
+def getallAnswer(submitted, pieces, leading_code, trailing_code):
+    if submitted == '':
+        return ''
+    answer = ''
+    for piece in submitted.split('-'):
+        piece_no, indent = piece.split(':')
+        indent = int(indent)
+        piece_no = int(piece_no)
+        answer = leading_code + '\n' + answer + ('    ' * indent) + pieces[piece_no] + '\n' + trailing_code + '\n'
+    return answer
+
+
+def gettarilAnswer(submitted, pieces, trailing_code):
+    if submitted == '':
+        return ''
+    answer = ''
+    for piece in submitted.split('-'):
+        piece_no, indent = piece.split(':')
+        indent = int(indent)
+        piece_no = int(piece_no)
+        answer = answer + ('    ' * indent) + pieces[piece_no] + '\n' + trailing_code + '\n'
+    return answer
+
+
+def getleadAnswer(submitted, pieces, leading_code):
+    if submitted == '':
+        return ''
+    answer = ''
+    for piece in submitted.split('-'):
+        piece_no, indent = piece.split(':')
+        indent = int(indent)
+        piece_no = int(piece_no)
+        answer = leading_code + '\n' + answer + ('    ' * indent) + pieces[piece_no] + '\n'
+    return answer
```

Comment:
These three functions all look almost identical, can we refactor so that we only have one copy of the logic?